packages/cli/src/module/generate-routes.ts

```diff
@@ -31,21 +32,28 @@ async function getRouteGenerator<Config extends ExtendedRoutesConfig>(metadata:
   // default route generator for express/koa/hapi
   // custom route generator
   if (routesConfig.routeGenerator !== undefined) {
-    try {
-      // try as a module import
-      const module = await import(routesConfig.routeGenerator);
-      return new module.default(metadata, routesConfig);
-    } catch (_err) {
-      // try to find a relative import path
-      const relativePath = path.relative(__dirname, routesConfig.routeGenerator);
-      const module = await import(relativePath);
-      return new module.default(metadata, routesConfig);
+    if (typeof routesConfig.routeGenerator === 'string') {
```

Comment:
TS doesn't properly discriminate on object props. Assign it `const routeGenerator = routesConfig.routeGenerator)` and the discrimination should work